src/ObfDataInterface.cpp

```diff
@@ -390,21 +387,20 @@ bool OsmAnd::ObfDataInterface::loadMapObjects(
 
             const auto& obfInfo = obfReader->obtainInfo();
 
-            // Skip all OBF readers that have map section, since they were already processed
-            if (!obfInfo->mapSections.isEmpty())
-                continue;
-
             for (const auto& routingSection : constOf(obfInfo->routingSections))
             {
                 // Check if request is aborted
                 if (queryController && queryController->isAborted())
                     return false;
 
-                // Check that map section with same name was not processed from other file
-                if (processedMapSectionsNames.contains(routingSection->name))
+                // Check that routing section with same name was not processed from other file
```

Comment:
strange that it doesn't differentiate where routing & map section comes from ".road.obf" or ".obf". In case it comes from  ".obf" we duplicate data! In  case from ".road.obf" we need to read from both sections & merge.